src/controllers/validate.controller.ts

```diff
@@ -0,0 +1,69 @@
+import { NextFunction, Request, Response, Router } from 'express';
+
+import { Controller } from '../interfaces';
+
+import { ParserService } from '../services/parser.service';
+
+import { ProblemException } from '../exceptions/problem.exception';
+import { prepareParserConfig, tryConvertToProblemException } from '../utils/parser';
+
+const BAD_REQUEST = 'Bad Request';
+
+/**
+ * Controller which exposes the Parser functionality, to validate the AsyncAPI document.
+ */
+export class ValidateController implements Controller {
+  public basepath = '/validate';
+
+  private parserService = new ParserService();
+
+  private async validate(req: Request, res: Response, next: NextFunction) {
+    try {
+      const contentType = req.headers['content-type'];
+
+      if (!contentType || (contentType.indexOf('application/json') !== 0 && contentType.indexOf('application/x-yaml') !== 0)) {
```

Comment:
I have problem with that check, because if we go with that direction, then you can send AsyncAPI document as YAML in the `req.body` (without any nesting) and if with JSON with nested field `asyncapi`:

```json
{
  "asyncapi": ...document
}
```

so as I wrote in my previous comment, we should go only with `asyncapi` field or without that field and treat whole body as AsyncAPI document :)

Reply:
this makes perfect sense, thank you 😃. I think we should go only with `asyncapi`, and if in the future we need to extend this endpoint's API, we can add new optional fields without breaking changes 👍 .... but writing that made me think. We could also extend creating a new endpoint, so maybe it's not a bad idea to simply treat the whole body as an AsyncAPI document.

I'll implement with the `asyncapi` field, what do you think @magicmatatjahu  and @smoya ?

Reply:
wrote some concerns in https://github.com/asyncapi/server-api/pull/22#issuecomment-998193256

Reply:
I'm attempting to remove the field and treat te whole body as an AsyncAPI document, but `request-body-validation.middleware` is throwing an error "RangeError: Maximum call stack size exceeded".
The openapi.yml is like this now:

```yaml
/validate:
    post:
      summary: Validate the given AsyncAPI document.
      operationId: validate
      tags:
        - validate
      requestBody:
        description: Validate the given AsyncAPI document with the AsyncAPI parser.
        required: true
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/AsyncAPIDocument'
          application/x-yaml:
            schema:
              $ref: '#/components/schemas/AsyncAPIDocument'
```

@magicmatatjahu or @smoya any idea what could fix this? I think the problem comes from the fact that the AsyncAPIDocument can be of type `https://raw.githubusercontent.com/asyncapi/spec-json-schemas/master/schemas/2.0.0.json`.

Reply:
I ended up skipping the validation from this middleware for the `validate` controller. It seems there is a problem for that AsyncAPI schema in the ajv module. I dug around and found this issue opened: https://github.com/ajv-validator/ajv/issues/1705

Also, I ran some tests and it does seem we have some serious nesting on the `oneOf` part of the schema, at a certain point the path inside ajv becomes something like this `.../definitions/schemaArray/items/definitions/schemaArray/items/definitions/schemaArray/items/definitions/schemaArray/items/definitions/schemaArray/items/definitions/schemaArray/items/definitions/...`

Let me know what you think 👍

Reply:
Yes, there are some "errors" related to the circular refs and due to that fact AJV throws errors and stackoverflow. Hmm.. I think that we should move to the old implementation with `asyncapi` field to simplify solution and have that benefit with easier future changes. Sorry my previous suggestion, it was wrong :(